GridControlDescriptor_Serialization / GridControlDataDto_Serialization tests fail with ECJ-embedded compile errors at runtime

## Summary

Running `mvn package` (or `mvn test`) in `webprotege-gwt-ui-server-core` fails two tests with a `java.lang.Error: Unresolved compilation problems:` thrown at test-method entry, even though the test sources compile cleanly with `javac`:

```
[ERROR] shouldSerialize(edu.stanford.bmir.protege.web.shared.form.field.GridControlDescriptor_Serialization_TestCase)
java.lang.Error:
Unresolved compilation problems:
    The method get(ImmutableList<GridColumnDescriptor>, FormSubjectFactoryDescriptor) in the type
        GridControlDescriptor is not applicable for the arguments
        (ImmutableList<GridColumnDescriptor>, int, FormSubjectFactoryDescriptor)
    The method getJson() is undefined for the type Object
    The method hasJsonPathArrayValue(String) is undefined for the type ObjectAssert<Object>
    The method hasJsonPath(String) is undefined for the type ObjectAssert<Object>
```

Affected tests:

- `webprotege-gwt-ui-server-core/src/test/java/edu/stanford/bmir/protege/web/shared/form/field/GridControlDescriptor_Serialization_TestCase.java`
- `webprotege-gwt-ui-server-core/src/test/java/edu/stanford/bmir/protege/web/shared/form/data/GridControlDataDto_Serialization_TestCase.java`

## Root cause

The `java.lang.Error: Unresolved compilation problems:` signature is unique to the **Eclipse JDT compiler (ECJ)** — `javac` never produces such class files; it fails the build on errors instead. ECJ emits a `.class` file even when the method body has compile errors, and embeds a `throw new Error(...)` at the erroring call site.

Inspecting the stale `.class` confirms ECJ origin:

```
$ javap -v webprotege-gwt-ui-server-core/target/test-classes/.../GridControlDescriptor_Serialization_TestCase.class
  Last modified Apr 17, 2026; size 3894 bytes
  ...
  #118 = String  "Unresolved compilation problems: \n\tThe method get(...) ..."
```

The error string is literally in the class file's constant pool.

The source itself is correct:

- It imports `org.springframework.boot.test.json.JacksonTester`; `tester.write(...)` returns `JsonContent<T>`, whose `AssertProvider` surfaces `JsonContentAssert` with `hasJsonPath(...)` / `hasJsonPathArrayValue(...)`.
- It calls the current 3-argument `GridControlDescriptor.get(columns, int pageSize, factory)` introduced in commit `e3d8bef22` (2025-07-14, "Support specification of page size in the UI").

Under ECJ with an incomplete project classpath — e.g. IntelliJ/Eclipse indexing the project before `spring-boot-test` is resolved, or using a cached older `webprotege-gwt-ui-shared` where `GridControlDescriptor.get` still had the 2-arg signature — both the method-resolution and the `JsonContent` type-inference fail, and the errors get baked into the `.class`.

## Why `mvn package` doesn't overwrite the stale file

`maven-compiler-plugin` runs in incremental mode by default: it compares source mtime against class mtime. The ECJ-produced `.class` has a later mtime than the `.java`, so Maven skips recompilation and surefire picks up the poisoned file. Confirmed by `stat`:

```
source: Apr 16 14:43:58 2026  GridControlDescriptor_Serialization_TestCase.java
class:  Apr 17 17:03:**       GridControlDescriptor_Serialization_TestCase.class  ← later
```

## Proposed fix

Two orthogonal actions:

1. **Immediate workaround:** document that `mvn clean` must be run before `mvn test` / `mvn package` when switching from an IDE build to the CLI. `mvn clean package` reproducibly succeeds — only dirty incremental builds show this failure.

2. **Real fix (pick one or more):**
   - Configure the IDE to delegate build & run actions to Maven (IntelliJ: *Build, Execution, Deployment → Build Tools → Maven → Runner → "Delegate IDE build/run actions to Maven"*). Then `target/test-classes` is always `javac`-produced and the failure mode is impossible.
   - Add a CI / pre-push check that greps `target/test-classes` for `"Unresolved compilation problems"` and fails if any `.class` contains it.
   - Optionally set `<useIncrementalCompilation>false</useIncrementalCompilation>` on `maven-compiler-plugin` for the `testCompile` goal in `webprotege-gwt-ui-server-core/pom.xml` so stale classes are always overwritten, at the cost of slower test builds.

## Not a regression from the dependency upgrade

The test `.class` file timestamp (2026-04-17 17:03) is **after** the Dagger/AutoValue/AutoFactory upgrade was applied on branch `fix/dagger-autovalue-autofactory-upgrade`. The IDE re-compiled the test in the background against a classpath that didn't yet reflect the refreshed artifacts. This issue is independent of the dep bump and would reproduce on `master` the same way if an IDE were driving the compile.

## Verification

1. `mvn clean test -pl webprotege-gwt-ui-server-core` — expect both tests to pass.
2. `javap -v target/test-classes/.../GridControlDescriptor_Serialization_TestCase.class | grep -c "Unresolved compilation problems"` — expect `0`.
3. Without cleaning, trigger an IDE build, then run `mvn test` — reproduce the failure, confirm `mvn clean test` fixes it.

## Environment

- Branch: `fix/dagger-autovalue-autofactory-upgrade` (commit `ec9240c69`).
- JDK 22, Maven 3.9.9.
- Affected tests added in commit `e3d8bef22` (2025-07-14).

--- a/webprotege-gwt-ui-server-core/pom.xml
+++ b/webprotege-gwt-ui-server-core/pom.xml
@@ -154,6 +154,32 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-clean-plugin</artifactId>
+                <executions>
+                    <!--
+                        Wipe target/test-classes before each test-compile so an IDE's
+                        incremental ECJ output (which can bake "Unresolved compilation
+                        problems" Errors into .class files with a newer mtime than the
+                        source) cannot survive a Maven build. See #265.
+                    -->
+                    <execution>
+                        <id>clean-test-classes</id>
+                        <phase>process-test-sources</phase>
+                        <goals>
+                            <goal>clean</goal>
+                        </goals>
+                        <configuration>
+                            <excludeDefaultDirectories>true</excludeDefaultDirectories>
+                            <filesets>
+                                <fileset>
+                                    <directory>${project.build.testOutputDirectory}</directory>
+                                </fileset>
+                            </filesets>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
         </plugins>
     </build>
 

